### DISCUSSION TITLE
Error in formr::formr_results()

### DISCUSSION THREAD

**Discussion by dfeistauer** [rubenarslan/formr.org#515] [Q&A] [answered]
Hi,

I try to download data from formr via R. In December it was working, now I get this error.

`formr::formr_store_keys(account_name = "fe...@uni-muenster.de")`
Enter your email: fe...@uni-muenster.de

`immer_4 <- formr::formr_results(survey_name = "immer_4", host = "https://formr.uni-muenster.de")`
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE html> <html>     <hea
                     (right here) ------^

I checked my Password, I installed `devtools::install_github("rubenarslan/formr")` new. But nothing helps with the problem. The error is always the same.

I could use export results via the webpage. However than I don't get the labels, ...
Any idea, how I get my data?

Thanks
Daniela

**Comment by rubenarslan** [comment]
Since you didn't show this in your example, could it be that you're missing this step?

`formr::formr_connect(keyring = "fe...@uni-muenster.de", host = "https://formr.uni-muenster.de")`

**Comment by dfeistauer** [comment]
you are right. Thanks. But I still get the same error.

**Comment by rubenarslan** [comment]
Please post your complete script until the formr_results part (minus the accounts/passwords of course) and the error messages/warnings you get at each step.

**Comment by dfeistauer** [comment]
```
> if(!require(devtools)) install.packages("devtools")
Loading required package: devtools
Loading required package: usethis
> if(!require(formr)) devtools::install_github("rubenarslan/formr")
Loading required package: formr
> # 2. Verbindung zu Formr herstellen ####
>   formr::formr_store_keys(account_name = "fe...@uni-muenster.de")
Enter your email:   formr::formr_connect(keyring = "fe...@uni-muenster.de", host = "https://formr.uni-muenster.de")
>   immer_4 <- formr::formr_results(survey_name = "KiEmpra_immer_4", host = "https://formr.uni-muenster.de")
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE html> <html>     <hea
                     (right here) ------^
> 
```

**Comment by rubenarslan** [comment]
OK. So, the formr_store_keys function is interactive. You're supposed to run it once to save your password. It's interactive, so that the password doesn't need to show up in your script. Once you've stored your keys, you can comment this part out. 
I think what's happening here is that R is taking the command right after (formr_connect) as input to the password prompt, so formr_connect is never run. Try running the code line-by-line. You might have to correct the stored password.

**Comment by dfeistauer** [comment]
I restartet RStudio.

Tried only the formr::formr_store_keys() and after trying to download the data I got the same error as before.
```
> # 2. Verbindung zu Formr herstellen ####
>   formr::formr_store_keys(account_name = "fe...@uni-muenster.de")
Enter your email: fe...@uni-muenster.de
>   immer_4 <- formr::formr_results(survey_name = "KiEmpra_immer_4",
+                                           host = "https://formr.uni-muenster.de")
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE html> <html>     <hea
                     (right here) ------^
```
This way did work in December. But not in January anymore.

After this I tried, formr::formr_connect() and gave again my password
```
>   formr::formr_connect(keyring = "fe...@uni-muenster.de", host = "https://formr.uni-muenster.de")
Error in formr::formr_connect(keyring = "fe...@uni-muenster.de", host = "https://formr.uni-muenster.de") : 
  Could not login for unknown reason.
```

**Comment by rubenarslan** [comment, marked as answer]
Hi Daniela, you're getting a different error message now, you're not getting logged in. 

I suspect you've stored the wrong credentials using formr_store_keys. You could just change your keyname to something else and make sure you store the correct email and password in there. After you've stored your keys, you should not have to run this command formr_store_keys again, unless you change your password. Thereafter, you should only need

formr_connect(keyring = "XXX")
formr_results("survey_name")

I tried accessing your server using my account and it worked as intended, so there shouldn't be any server-side reason.

**Comment by dfeistauer** [comment]
ah, ok. I thought I always need to give the email and password.

```
# 2. Verbindung zu Formr herstellen ####
  #formr::formr_store_keys(account_name = "ZugangFormrFeistauer")
  formr::formr_connect(keyring = "ZugangFormrFeistauer", host = "https://formr.uni-muenster.de")
```

Now it works. Thanks
